### PR TITLE
CHAIN-557 extend low gas alert

### PR DESCRIPTION
**Pull request by imykolenko**
extend gas monitor to cover testnet challenge airdropoer gas/usdc and bitcoin fee payer

### CODE REVIEW

File: backend/src/main/kotlin/xyz/funkybit/core/repeater/tasks/BalancesMonitorTask.kt [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+                    }
+                }
+            }
+        } finally {
+            nextInvocationArgs = null
+            HttpClient.setQuietModeForThread(false)
+        }
+    }
+
+    private fun checkBitcoinFeePayerGas(minimumGas: BigInteger = BigInteger.valueOf(100_000)) {
```
Comment (bholzman) [inline backend/src/main/kotlin/xyz/funkybit/core/repeater/tasks/BalancesMonitorTask.kt:72]: should these defaults be in the config? I'm also thinking we might want a bit of a higher minimum here, maybe more like 0.1 or 0.05 btc, and maybe more like 500 or 1000 minEnrollments for testnet challenge

**Review by bholzman — APPROVED** [review]
Looks great!

File: backend/src/main/kotlin/xyz/funkybit/core/repeater/tasks/BalancesMonitorTask.kt [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+import xyz.funkybit.core.model.db.SymbolEntity
+import xyz.funkybit.core.services.UtxoManager
+import xyz.funkybit.core.utils.HttpClient
+import xyz.funkybit.core.utils.TestnetChallengeUtils
+import xyz.funkybit.core.utils.toFundamentalUnits
+import java.math.BigInteger
+import kotlin.time.Duration.Companion.minutes
+
+data class BalancesMonitorConfig(
+    val testnetChallengeEnabled: Boolean = System.getenv("TESTNET_CHALLENGE_ENABLED")?.toBoolean() ?: true,
```
[the remark this answers is not in the document]
Comment (imykolenko) [reply backend/src/main/kotlin/xyz/funkybit/core/repeater/tasks/BalancesMonitorTask.kt:18]: thx!

File: backend/src/main/kotlin/xyz/funkybit/core/repeater/tasks/BalancesMonitorTask.kt (hunk 1 above)
Comment (imykolenko) [reply backend/src/main/kotlin/xyz/funkybit/core/repeater/tasks/BalancesMonitorTask.kt]: good point! Moved to config, bumped defaults to 0.1 btc and 1000 min enrollements.

**Pull request merged**